gradle test not worlking

Working to try this workshop with gradle, but looks like the build is broken. 
Im on Windows runnning `gradlew test`: 

> Exception in thread "main" java.lang.BootstrapMethodError: java.lang.NoSuchMethodError: org.junit.platform.engine.TestDescriptor.prune()V
>         at org.junit.platform.launcher.core.Root.prune(Root.java:71)
>         at org.junit.platform.launcher.core.DefaultLauncher.discoverRoot(DefaultLauncher.java:123)
>         at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:87)
>         at org.junit.platform.console.tasks.ConsoleTestExecutor.executeTests(ConsoleTestExecutor.java:65)
>         at org.junit.platform.console.tasks.ConsoleTestExecutor.lambda$execute$0(ConsoleTestExecutor.java:57)
>         at org.junit.platform.console.tasks.CustomContextClassLoaderExecutor.invoke(CustomContextClassLoaderExecutor.java:33)
>         at org.junit.platform.console.tasks.ConsoleTestExecutor.execute(ConsoleTestExecutor.java:57)
>         at org.junit.platform.console.ConsoleLauncher.executeTests(ConsoleLauncher.java:79)
>         at org.junit.platform.console.ConsoleLauncher.execute(ConsoleLauncher.java:69)
>         at org.junit.platform.console.ConsoleLauncher.main(ConsoleLauncher.java:41)
> Caused by: java.lang.NoSuchMethodError: org.junit.platform.engine.TestDescriptor.prune()V
>         at java.lang.invoke.MethodHandleNatives.resolve(Native Method)
>         at java.lang.invoke.MemberName$Factory.resolve(MemberName.java:975)
>         at java.lang.invoke.MemberName$Factory.resolveOrFail(MemberName.java:1000)
>         at java.lang.invoke.MethodHandles$Lookup.resolveOrFail(MethodHandles.java:1392)
>         at java.lang.invoke.MethodHandles$Lookup.linkMethodHandleConstant(MethodHandles.java:1748)
>         at java.lang.invoke.MethodHandleNatives.linkMethodHandleConstant(MethodHandleNatives.java:477)
>         ... 10 more
>

--- a/README.md
+++ b/README.md
@@ -32,6 +32,6 @@ JUnit-5-specific plugins for **Maven** Surefire and **Gradle** are included in t
 Try it with `mvn clean test` or `gradle test`. Alternatively, you can use the provided Gradle or Maven wrappers with `./mvnw clean test` or `./gradlew test`.
 
 **IntelliJ** is the first (and so far only) tool that has native support and everything should work out of the box.
-Since this project uses JUnit 5 Milestone 3 [2016.3.1](https://blog.jetbrains.com/idea/2016/12/intellij-idea-2016-3-1-rc-updates-junit-5-support-to-m3/) is required to run the tests.
+Since this project uses JUnit 5 Milestone 4 [2017.1.2](https://blog.jetbrains.com/idea/2017/04/intellij-idea-2017-1-2-update-is-available/) is required to run the tests.
 
 **Eclipse** requires a little fiddling to get to work but it is possible with 4.7 M4, following [these steps](https://bugs.eclipse.org/bugs/show_bug.cgi?id=488566#c8).

--- a/build.gradle
+++ b/build.gradle
@@ -7,7 +7,7 @@ buildscript {
 		mavenCentral()
 	}
 	dependencies {
-		classpath 'org.junit.platform:junit-platform-gradle-plugin:1.0.0-M3'
+		classpath 'org.junit.platform:junit-platform-gradle-plugin:1.0.0-M4'
 	}
 }
 
@@ -32,14 +32,14 @@ dependencies {
 	// for writing tests "testCompile" scope would suffice, but extensions
  	// are defined in the project's "main" folder, so we need "compile"
 	// JUnit Jupiter API and TestEngine implementation
-	compile("org.junit.jupiter:junit-jupiter-api:5.0.0-M3")
+	compile("org.junit.jupiter:junit-jupiter-api:5.0.0-M4")
 
 	// other test libraries work out of the box. e.g. AssertJ and Mockito
 	testCompile("org.assertj:assertj-core:3.3.0")
 	testCompile("org.mockito:mockito-core:1.10.19")
 
 	// contains the engine that actually runs the Jupiter-tests
-	testRuntime("org.junit.jupiter:junit-jupiter-engine:5.0.0-M3")
+	testRuntime("org.junit.jupiter:junit-jupiter-engine:5.0.0-M4")
 }
 
 task wrapper(type: Wrapper) {

--- a/pom.xml
+++ b/pom.xml
@@ -12,7 +12,7 @@
 		<dependency>
 			<groupId>org.junit.jupiter</groupId>
 			<artifactId>junit-jupiter-api</artifactId>
-			<version>5.0.0-M3</version>
+			<version>5.0.0-M4</version>
 			<!-- for writing tests "test" scope would suffice,
 				but extensions are defined in the project's "main" folder, so we need "compile" -->
 			<scope>compile</scope>
@@ -55,13 +55,13 @@
 					<dependency>
 						<groupId>org.junit.platform</groupId>
 						<artifactId>junit-platform-surefire-provider</artifactId>
-						<version>1.0.0-M3</version>
+						<version>1.0.0-M4</version>
 					</dependency>
 					<dependency>
 						<!-- contains the engine that actually runs the Jupiter-tests -->
 						<groupId>org.junit.jupiter</groupId>
 						<artifactId>junit-jupiter-engine</artifactId>
-						<version>5.0.0-M3</version>
+						<version>5.0.0-M4</version>
 					</dependency>
 				</dependencies>
 			</plugin>

--- a/slides/05_modularization/modularization.md
+++ b/slides/05_modularization/modularization.md
@@ -68,7 +68,7 @@ You can run “Vintage” JUnit 4 tests with the JUnit Platform using the JUnit 
         <dependency>
             <groupId>org.junit.vintage</groupId>
             <artifactId>junit-vintage-engine</artifactId>
-            <version>4.12.0-M3</version>
+            <version>4.12.0-M4</version>
         </dependency>
     </dependencies>
 </plugin>

--- a/src/test/java/org/junit/junit5workshop/_1_basics/_5_NestedAndNamedTest.java
+++ b/src/test/java/org/junit/junit5workshop/_1_basics/_5_NestedAndNamedTest.java
@@ -19,7 +19,7 @@ class _5_NestedAndNamedTest {
 	/*
 	 * Demonstration of JUnit 5 features (particularly nesting and naming)
 	 * that was taken from the JUnit 5 User Guide
-	 * (http://junit.org/junit5/docs/5.0.0-M3/user-guide/#writing-tests-nested)
+	 * (http://junit.org/junit5/docs/5.0.0-M4/user-guide/#writing-tests-nested)
 	 */
 
 	Stack<Object> stack;

